File: cs50/week6/pset6/readability/readability.py
from cs50 import get_string
from cs50 import get_int



def count_letters(text):

    letter_count = 0

    for char in text:
        if char.isalpha():
            letter_count = letter_count + 1

    return letter_count


def count_words(text):

    word_count = 0

    for char in text:
        if char == " ":
            word_count = word_count + 1

    word_count = word_count + 1

    return word_count

def count_sentences(text):

    sentence_count = 0

    for char in text:
        if char == "." or char == "!" or char == "?":
            sentence_count = sentence_count + 1

    return sentence_count




text = get_string("Enter text: ")

#index = 0.0588 * L - 0.296 * S - 15.8
#L is the average number of letters per 100 words in the text
#S is the average number of sentences per 100 words in the text.

l_count = count_letters(text)
w_count = count_words(text)
s_count = count_sentences(text)


# print(str(l_count) + " letters")
# print(str(w_count) + " words")
# print(str(s_count) + " sentences")

L = (l_count / w_count) * 100
S = (s_count / w_count) * 100

index =  round(0.0588 * L - 0.296 * S - 15.8)

# print(round(index))

if index < 1:
    print("Before Grade 1")

elif index >= 1 and index < 16:
    print("Grade", str(index))

else:
    print("Grade 16+")

File: cs50/week2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string);
int count_words(string);
int count_sentences(string);

int main(void)
{

    string text = get_string("Text: ");
    float index;
    float l;
    float s;

    //printf("%s\n",text);

    int letter_count = count_letters(text);

    //printf("%i letter(s)\n", letter_count);

    int word_count = count_words(text);

    //printf("%i word(s)\n", word_count);

    int sentence_count = count_sentences(text);

    //printf("%i sentence(s)\n", sentence_count);


    l = ((float) letter_count / (float) word_count) * 100; // average number of letters per 100 words

    s = ((float) sentence_count / (float) word_count) * 100;

    //printf("%f %f\n", l, s);


    index = 0.0588 * l  - 0.296 * s - 15.8;

    index =  round(index);



    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) index);
    }





}

int count_letters(string text)
{
    int letter_count = 0;
    //printf("%s\n",text);
    size_t length = strlen(text);

    //printf("Length of text is %i\n", (int) length);

    for (int i = 0; i < length; i ++)
    {
        if (isalpha(text[i]))
        {
            letter_count ++;
        }
    }

    return letter_count;
}

int count_words(string text)
{
    int word_count = 0;

    size_t length = strlen(text);

    for (int i = 0; i < length; i ++)
    {
        if (text[i] == ' ')
        {
            word_count ++;
        }
    }

    word_count ++; // end of loop means we got to last word but there is no space to signify this

    return word_count;

}

int count_sentences(string text)
{

    int sentence_count = 0;

    size_t length = strlen(text);

    for (int i = 0; i < length; i ++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentence_count ++;
        }
    }

    return sentence_count;
}